HIP/hip-850.md

```diff
@@ -0,0 +1,70 @@
+---
+hip: 850
+title: Enhancing Supply Key Functionality for NFT Updates in Treasury Account
+author: Patches (@HGraphPunks)
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Last Call
+last-call-date-time: 2024-04-03T07:00:00Z
+created: 2024-01-02
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/660
+requested-by: HGraphPunks, TierBot
+updated: 2024-03-20
+---
+
+## Abstract
+
+This HIP proposes enhancing the Supply Key functionality within the Hedera ecosystem, specifically enabling the Supply Key to update an NFT metadata field while the NFT is held in the treasury account. This ensures secure and controlled adaptability of NFTs, meeting dynamic needs while maintaining their integrity post-distribution.
+
+## Motivation
+
+Currently, there is a limitation in dynamically updating NFTs in a controlled manner. The proposed enhancement to the Supply Key functionality addresses this gap, allowing for secure and selective updates of NFTs within the treasury account, a feature particularly relevant for various evolving use cases in the Hedera ecosystem.
```

Comment:
I still thinks it's a bit confusing that the Supply Key is used for this extra functionality.
Supply Key implies control over the supply quantity. This change the role description to include the attributes of the supply also i.e. metadata.

Have we considered building upon HIP 657 instead and utilizing the metadata key and a separate property for immutability when in the treasury?

Reply:
Yes, I had a conversation about this nuance earlier last week. The issue is that we have about 3 HIPs talking about different instances of metadata and it's complicating the discussions as there's shared terminology that are distinctly different. 

HIP-765 is not referencing the metadata on a NFT id. It is only referencing the metadata field being added to the token level of a token. An admin key will not be able to adapt the metadata of NFT id.

The admin key will be able to adapt the metadata field on the token level, like any other field on the token level.

Reply:
Comments further down expanded on details here.